docs/guide/arrays.md

```diff
@@ -0,0 +1,79 @@
+- Constrain the output array's size
+- Filter an array based on boolean arrays
+
+## About arrays in HyperFormula
+
+There are two types of arrays in HyperFormula:
+* **Range array**: a range of cell addresses (e.g. `A1:A10`)
+* **Ad-hoc array**: an array with no ground-truth range that defines the values (e.g. `5*A1:B5`)
+
+Non-array values are **scalars**.
+
+By default, array arithmetic is disabled globally in HyperFormula.
+
+To use array arithmetic, either enable it globally ([set the `useArrayArithmetic` option to `true`](../api/interfaces/configparams.html#usearrayarithmetic)), or use the ARRAYFORMULA function.
+
+## Operating on arrays
+
+When you perform an operation on an array, each output array value is a result of that operation on the corresponding input array value.
+
+To enable array arithmetic for your formula, use the [`ARRAYFORMULA`](built-in-functions.md#array-manipulation) function:
+
+| Syntax | Example |
+| :--- | :--- |
+| `ARRAYFORMULA(your_formula)` | `=ARRAYFORMULA(A2:A5*B2:B5)` |
+
+If you don't use the `ARRAYFORMULA` function, HyperFormula treats [ad-hoc arrays](#about-arrays-in-hyperformula) as [scalars](#about-arrays-in-hyperformula), taking only your ad-hoc array's top-left value.
+
+To enable array arithmetic globally, set the `useArrayArithmetic` option to `true` in HyperFormula [configuration](../api/interfaces/configparams.html#usearrayarithmetic).
+
+## Passing arrays to functions
+
+When you pass an array as a function's argument, the function returns an array as well.
+
+To pass an array as an argument, enable the array arithmetic with the [`ARRAYFORMULA`](built-in-functions.md#array-manipulation) function:
+
+| Syntax | Example |
+| :--- | :--- |
+| `YOUR_FUNCTION(ARRAYFORMULA(your_formula))` | `=ISEVEN(ARRAYFORMULA(A2:A5*10))` |
+
+If you don't use the `ARRAYFORMULA` function, HyperFormula treats [ad-hoc arrays](#about-arrays-in-hyperformula) as [scalars](#about-arrays-in-hyperformula), taking only your ad-hoc array's top-left value.
+
+To enable array arithmetic globally, set the `useArrayArithmetic` option to `true` in HyperFormula [configuration](../api/interfaces/configparams.html#usearrayarithmetic).
+
+## Constraining the output array's size
+
+To constrain the size of the output array, use the `ARRAY_CONSTRAIN` function:
+
+| Syntax | Example |
+| :--- | :--- |
+| `ARRAY_CONSTRAIN(your_array,your_height,your_width)` | `=ARRAY_CONSTRAIN(ARRAYFORMULA(A2:A5*10),2,1)` |
+
+If your specified output array size is smaller than the input array size, only the corresponding top-left cells of the input array are taken into account.
+
+If your specified output array size is larger or equal to the input array size, no change is made.
+
+## Filtering an array
+
+To filter an array based on boolean arrays, use the `FILTER` function:
+
+| Syntax | Example |
+| :--- | :--- |
+| `FILTER(your_array,BoolArray1[; BoolArray2[; ...]]` | `=FILTER(A2:A5*10),BoolArray1)` |
+
+## Array rules
```

Comment:
Below listed rules are rules for vectorization, which happens when function that expect scalars get vectors.

There are regular functions that accept arrays as input, and no such rules apply to those.

Reply:
@izulin - [done](https://github.com/handsontable/hyperformula/pull/741/commits/d161d5262b27709a53d9c78b9f2a529928757f76#diff-8124285aa27825ee9cad11034f206a42efb0923699c2872d25764e120df72530R73)